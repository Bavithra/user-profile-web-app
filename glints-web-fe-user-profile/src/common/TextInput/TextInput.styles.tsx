import styled from "styled-components";
import { Colors } from "../../styles/Colors";

type Props = {
  hasValue: boolean;
};

export const Label = styled.label`
  height: 12px;
  color: ${Colors.grey2};
  font-weight: normal;
  opacity: 0.75;
  order: 1;
  padding-left: 4px;
  pointer-events: none;
  text-shadow: none;
  text-transform: capitalize;
  transform-origin: left top;
  transform: scale(1) translate3d(0, 22px, 0);
  transition: 200ms ease all;

  ${(props: Props) =>
    props.hasValue
      ? `
      color: ${Colors.green1};
      opacity: 1;
      transform: scale(0.8) translate3d(0, -8px, 0);}`
      : ""}

`;

export const Input = styled.input`
  border-radius: 0;
  display: flex;
  font-size: 100%;
  line-height: 36px;
  text-shadow: none;
  padding-left: 4px;

  border: 0;
  border: 1px solid ${Colors.grey4};
  color: #000;
  flex: 1 1 auto;
  order: 2;


  &:focus {
    outline: 0;
  }

  &:not(:focus) {
    color: ${(props: Props) => (props.hasValue ? "" : "transparent")};
  }

  &:focus + ${Label} {
    color: ${Colors.green1};
    opacity: 1;
    transform: scale(0.8) translate3d(0, -8px, 0);
  }
`;
