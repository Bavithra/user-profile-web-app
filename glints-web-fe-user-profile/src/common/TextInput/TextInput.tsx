import React, { ChangeEvent } from "react";
import { MAX_TEXT_INPUT } from "../../Constants";
import { InputContainer } from "../../styles/Common.styles";
import { Input, Label } from "./TextInput.styles";

type Props = {
  label: string;
  onChange: (value: string) => void;
  type?: string;
  value?: string;
  readOnly?: boolean;
  onBlur?: () => void;
};

export default function TextInput(props: Props) {
  const {
    label,
    value,
    readOnly = false,
    type = "string",
    onChange,
    onBlur,
  } = props;

  function handleChange(e: ChangeEvent<HTMLInputElement>) {
    onChange(e.target.value);
  }

  function hasValue() {
    if (value) {
      return value?.length > 0;
    }

    return false;
  }

  return (
    <InputContainer>
      <Input
        readOnly={readOnly}
        type={type}
        value={value}
        hasValue={hasValue()}
        onChange={handleChange}
        onBlur={onBlur}
        maxLength={MAX_TEXT_INPUT}
        min="0"
      />
      <Label hasValue={hasValue()}>{label}</Label>
    </InputContainer>
  );
}
