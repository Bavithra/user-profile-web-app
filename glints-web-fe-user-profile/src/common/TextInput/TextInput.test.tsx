import React from "react";
import renderer from "react-test-renderer";
import TextInput from "./TextInput";

describe("TextInput Block", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(
        <TextInput
          label={""}
          onChange={function (value: string): void {
            throw new Error("Function not implemented.");
          }}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
