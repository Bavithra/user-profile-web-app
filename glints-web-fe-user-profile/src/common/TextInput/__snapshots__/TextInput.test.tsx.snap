// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput Block renders correctly 1`] = `
<div
  className="sc-dkzDqf fSUJtn"
>
  <input
    className="sc-eCYdqJ feQGCx"
    maxLength={200}
    min="0"
    onChange={[Function]}
    readOnly={false}
    type="string"
  />
  <label
    className="sc-hKMtZM jtjkLl"
  />
</div>
`;
